tests/regression/annotations.wast

```diff
@@ -0,0 +1,147 @@
+(@a , ; ] [ }} }x{ ({) ,{{};}] ;)
+(@a (bla) () (5-g) ("aa" a) ($x) (bla bla) (x (y)) ")" "(" x")"y)
+(@a @ @x (@x) (@x y) (@) (@ x) (@(@(@(@)))))
+(@a (;bla;) (; ) ;)
+  ;; bla)
+  ;; bla (@x
+)
+
+(assert_malformed (module quote "( @a)") "unknown operator")
+
+(assert_malformed (module quote "(@)") "malformed annotation id")
+(assert_malformed (module quote "(@ )") "malformed annotation id")
+(assert_malformed (module quote "(@ x)") "malformed annotation id")
+(assert_malformed (module quote "(@(@a)x)") "malformed annotation id")
+
+(assert_malformed (module quote "(@x ") "unclosed annotation")
+(assert_malformed (module quote "(@x ()") "unclosed annotation")
+(assert_malformed (module quote "(@x (y (z))") "unclosed annotation")
+(assert_malformed (module quote "(@x (@y )") "unclosed annotation")
+
+(assert_malformed (module quote "(@x))") "unexpected token")
+(assert_malformed (module quote "(@x ()))") "unexpected token")
+(assert_malformed (module quote "(@x (y (z))))") "unexpected token")
+(assert_malformed (module quote "(@x (@y )))") "unexpected token")
+
+(assert_malformed (module quote "(@x \"") "unclosed string")
+(assert_malformed (module quote "(@x \")") "unclosed string")
+
+(assert_malformed (module quote "((@a)@b)") "unknown operator")
+(assert_malformed (module quote "(func $(@a))") "unknown operator")
+(assert_malformed (module quote "(func $(@a)f)") "unknown operator")
+
+((@a) module (@a) $m (@a) (@a)
+  ((@a) import (@a) "spectest" (@a) "global_i32" (@a)
+    ((@a) global (@a) $g (@a) i32 (@a)) (@a)
+  ) (@a)
+  ((@a) import (@a) "spectest" (@a) "table" (@a)
+    ((@a) table (@a) $t (@a) 10 (@a) 20 (@a) funcref (@a)) (@a)
+  ) (@a)
+  ((@a) import (@a) "spectest" (@a) "memory" (@a)
+    ((@a) memory (@a) $m (@a) 1 (@a) 2 (@a)) (@a)
+  ) (@a)
+  ((@a) import (@a) "spectest" (@a) "print_i32_f32" (@a)
+    ((@a) func (@a) $f (@a)
+      ((@a) param (@a) i32 (@a) f32 (@a)) (@a)
+      ((@a) result (@a)) (@a)
+    ) (@a)
+  ) (@a)
+
+  ((@a) export (@a) "g" (@a)
+    ((@a) global (@a) $g (@a)) (@a)
+  ) (@a)
+  ((@a) export (@a) "t" (@a)
+    ((@a) table (@a) $t (@a)) (@a)
+  ) (@a)
+  ((@a) export (@a) "m" (@a)
+    ((@a) memory (@a) $m (@a)) (@a)
+  ) (@a)
+  ((@a) export (@a) "f" (@a)
+    ((@a) func (@a) $f (@a)) (@a)
+  ) (@a)
+) (@a)
+
+((@a) module (@a) $m (@a) (@a)
+  ((@a) global (@a) $g (@a)
+    ((@a) export (@a) "g" (@a)) (@a)
+    ((@a) import (@a) "spectest" (@a) "global_i32" (@a)) (@a)
+    i32 (@a)
+  ) (@a)
+  ((@a) table (@a) $t (@a)
+    ((@a) export (@a) "t" (@a)) (@a)
+    ((@a) import (@a) "spectest" (@a) "table" (@a)) (@a)
+    10 (@a) 20 (@a)
+    funcref (@a)
+  ) (@a)
+  ((@a) memory (@a) $m (@a)
+    ((@a) export (@a) "m" (@a)) (@a)
+    ((@a) import (@a) "spectest" (@a) "memory" (@a)) (@a)
+    1 (@a) 2 (@a)
+  ) (@a)
+  ((@a) func (@a) $f (@a)
+    ((@a) export (@a) "f" (@a)) (@a)
+    ((@a) import (@a) "spectest" (@a) "print_i32_f32" (@a)) (@a)
+    ((@a) param (@a) i32 (@a) f32 (@a)) (@a)
```

Comment:
:rofl:

Reply:
Heh and to be clear, this came from the spec repo, so credit goes elsewhere to whose idea this was